ec/ecresource/deploymentresource/read.go

```diff
@@ -92,3 +173,53 @@ func deploymentNotFound(err error) bool {
 	// We also check for the case where a 403 is thrown for ESS.
 	return apierror.IsRuntimeStatusCode(err, 403)
 }
+
+var minimumSupportedVersion = semver.MustParse("6.6.0")
+
+func checkVersion(version string) error {
+	v, err := semver.New(version)
+
+	if err != nil {
+		return fmt.Errorf("unable to parse deployment version: %w", err)
+	}
+
+	if v.LT(minimumSupportedVersion) {
+		return fmt.Errorf(
+			`invalid deployment version "%s": minimum supported version is "%s"`,
+			v.String(), minimumSupportedVersion.String(),
+		)
+	}
+
+	return nil
+}
+
+func HasRunningResources(res *models.DeploymentGetResponse) bool {
```

Comment:
```suggestion
func hasRunningResources(res *models.DeploymentGetResponse) bool {
```

Reply:
We cannot rename it - it's tested in `deploymentresource_test`. And the test package is named differently because it imports `github.com/elastic/terraform-provider-ec/ec/ecresource/deploymentresource`.

Reply:
So it would be a circular dependency if the test package was named `deploymentresource`.